### ISSUE TITLE
Support ASV and USV like CSV

### ISSUE THREAD

**Issue by ptman** [go-gitea/gitea#29761]
### Feature Description

CSV is already supported. Similar support for ASV and USV would be great. https://github.com/sixarm/usv

**Comment by silverwind** [comment]
There are numerous variants to CSV, like TSV (tabs), PSV (pipe). Maybe the separator characters should be configurable 😆.

**Comment by lunny** [comment]
> There are numerous variants to CSV, like TSV (tabs), PSV (pipe). Maybe the separator characters should be configurable 😆.

I think maybe we can have a guess.

**Comment by silverwind** [comment]
Yeah, some of these formats have well-known file extensions, so it should be possible.

**Comment by KN4CK3R** [comment]
I already have implemented this:
https://github.com/go-gitea/gitea/blob/712e19fa6fbf2f1a5b0a471782d38a7d91e538ae/modules/csv/csv.go#L53

May have to re-check when we try to create a csv reader.
